### ISSUE TITLE
Lower bid may become top bid

### ISSUE THREAD

**Issue by takenorim** [prebid/Prebid.js#9343]
## Type of issue
A bug

## Description
For some case, I found lower bid became the top bid keyvalues when sending keyvalue to GAM XHR. Those are actual examples. 

- Example 1 - `hb_pb_ogury=266.00` v.s `hb_pb_rubicon=7000.00`
```
...(snip)...
hb_format_rubicon=banner&
hb_size_rubicon=320x50&
hb_pb_rubicon=7000.00&
hb_adid_rubicon=95af1cb036c808a&
hb_bidder_rubicon=rubicon&
hb_format_ogury=banner&
hb_size_ogury=320x100&
hb_pb_ogury=266.00&
hb_adid_ogury=8954d33907616c5&
hb_bidder_ogury=ogury&
hb_format=banner&
hb_size=320x100&
hb_pb=266.00&
hb_adid=8954d33907616c5&
hb_bidder=ogury|
...(snip)...
```

- Example 2 - `hb_pb_yahoossp=5.00` v.s. `hb_pb_rubicon=7000.00`
```
...(snip)...
hb_format_yahoossp=banner&
hb_size_yahoossp=320x50&
hb_pb_yahoossp=5.00&
hb_adid_yahoossp=5d4b62d7ca8d43&
hb_bidder_yahoossp=yahoossp&
hb_format_rubicon=banner&
hb_size_rubicon=320x50&
hb_pb_rubicon=7000.00&
hb_adid_rubicon=99c790ecaccc719&
hb_bidder_rubicon=rubicon&
hb_format=banner&
hb_size=320x50&
hb_pb=5.00&
hb_adid=5d4b62d7ca8d43&
hb_bidder=yahoossp|
...(snip)...
```

## Steps to reproduce
I use Local Override and put some console.log for debugging. Long story in short, for some bidders the cpm is casting as type:strings, and other bidders use cpm as type:number. Then, the `utils.js#getHighestCpmCallback` does NOT work as expected. See the screenshot below.

![Screen Shot 2022-12-15 at 17 10 46](https://user-images.githubusercontent.com/19651358/207806697-75352002-5eab-4041-b979-f500e4074b27.jpg)

### Expected results
Highest bid becomes top bid.

### Actual results
For some special cases, lower bid becomes top bid (e.g. `hb_pb_yahoossp=5.00` wins against `hb_pb_rubicon=7000.00`, and top bid becomes `hb_pb=5.00`)

## Platform details
I tested with the PBJS v7.22.

## Other information
This is essentially something like below happened.

```
if ( '13736.3679' < '5.7143' ) {
    console.log('true');
} else {
    console.log('false')
}

true
```

Of course, the following works as expected.
```
if ( 13736.3679 < 5.7143 ) {
    console.log('true');
} else {
    console.log('false')
}

false
```

**Comment by takenorim** [comment]
## Additional information
In my case, I'm using the currency module to convert USD to JPY. For some bidders (e.g. rubicon), original CPM is USD, but it looks the currency module convert new cpm as type:strings by `toFixed()` method, see below.

- https://github.com/prebid/Prebid.js/blob/master/modules/currency.js#L237
```
          bid.cpm = (parseFloat(bid.cpm) * conversion).toFixed(4);
```

**Comment by takenorim** [comment]
Simple and safe fix would be using the `parseFloat` like below.

- https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L739-L746
```
function getHighestCpmCallback(useTieBreakerProperty, tieBreakerCallback) {
  return (previous, current) => {
    if (previous.cpm === current.cpm) {
      return tieBreakerCallback(previous[useTieBreakerProperty], current[useTieBreakerProperty]) ? current : previous;
    }
    return previous.cpm < current.cpm ? current : previous;
  }
}
```
->
```
function getHighestCpmCallback(useTieBreakerProperty, tieBreakerCallback) {
  return (previous, current) => {
    if (parseFloat(previous.cpm) === parseFloat(current.cpm) ) {
      return tieBreakerCallback(previous[useTieBreakerProperty], current[useTieBreakerProperty]) ? current : previous;
    }
    return parseFloat(previous.cpm) < parseFloat(current.cpm) ? current : previous;
  }
}
```

As long as I did simple testing with Chrome Local Override, this idea works. GAM key-value becomes like below, now the top bid hb_pb is the highest CPM bid.

```
...(snip)...
hb_format_yahoossp=banner&
hb_size_yahoossp=320x50&
hb_pb_yahoossp=5.00&
hb_adid_yahoossp=8307175c9e6ab64&
hb_bidder_yahoossp=yahoossp&
hb_format_rubicon=banner&
hb_size_rubicon=320x50&
hb_pb_rubicon=7000.00&
hb_adid_rubicon=9261fc01543bc05&
hb_bidder_rubicon=rubicon&
hb_format=banner&
hb_size=320x50&
hb_pb=7000.00&
hb_adid=9261fc01543bc05&
hb_bidder=rubicon|
...(snip)...
```

**Comment by dgirardi** [comment]
I believe this is a duplicate of https://github.com/prebid/Prebid.js/issues/9271, fixed in #9273 (since version 7.27).

**Comment by takenorim** [comment]
Thanks for that prompt response. Ah, I did not notice #9271. Yes, this is a duplicated issue. Let me close this issue.
